### PR TITLE
Support calibration benches on apple silicon macs

### PR DESCRIPTION
**Pull request by graydon**
This adds support for running the budget calibration benchmark code on M1 / apple silicon macs.

(It might also work on intel macs, I will give it a try tomorrow morning, but this should be enough to unblock @jayz22 on calibration)

### CODE REVIEW

File: soroban-env-host/Cargo.toml [hunk 1]
```diff
@@ -40,9 +43,10 @@ testutils = []
 
 [target.'cfg(target_os = "linux")'.dev-dependencies]
 perf-event = "0.4.7"
-tracking-allocator = "0.4.0"
-tabwriter = "1.2.1"
-thousands = "0.2.0"
+
+[target.'cfg(all(target_os = "macos", target_arch = "aarch64"))'.dev-dependencies]
+macos-perf = "0.1.1"
+sudo = "0.6.0"
```
Comment (leighmcculloch) [inline soroban-env-host/Cargo.toml:49]: The description of the `sudo` crate is terrifying. From the docs.rs:
>Detect if you are running as root, restart self with sudo if needed or setup uid zero when running with the SUID flag set.

As far as I can tell we're only using the detection mechanism, and not the auto-restart as sudo. Right?

**Review by leighmcculloch — APPROVED** [review]
One question, otherwise this is exciting 👏🏻.

**Comment by jayz22** [comment]
Thank you so much @graydon !

File: soroban-env-host/Cargo.toml (hunk 1 above)
Comment (graydon) [reply soroban-env-host/Cargo.toml:49]: Yeah. I think I will actually be able to roll this change back and use getrusage_v4 instead, but as it stands this is "use for detection only" and "only in a benchmark" -- no need to be alarmed.

**Pull request merged**